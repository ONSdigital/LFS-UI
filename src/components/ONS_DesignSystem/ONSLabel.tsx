import React, {Component} from 'react';

interface Props{
  label: string,
  description? : string,
  id?: string
}

export class ONSLabel extends Component <Props, {}>{
  constructor(props : Props) {
    super(props);
  }

    render() {
    return (
      <label className="label" id={this.props.id}>{this.props.label}
      {
        this.props.description !== undefined &&
        [<br/>,
        <span className="label__description">{this.props.description}</span>]
      }
      </label>
    );
  }
}
