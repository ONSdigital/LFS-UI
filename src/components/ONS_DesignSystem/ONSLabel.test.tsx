import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSLabel} from "./ONSLabel";
import Adapter from "enzyme-adapter-react-16";

describe("ONS Label Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {
        label: "ONS Label",
        description: "This is the ONS Label",
        id: "L1"
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSLabel
                label={props.label}
                description={props.description}
                id={props.id}>
            </ONSLabel>
            
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, Props).find("ONSLabel").getElement().props.label).toEqual(Props.label);
    })

    it('displays a description if supplied', () => {
        expect(wrapper(mount, Props).find("span").hasClass("label__description")).toEqual(true)
    })
})