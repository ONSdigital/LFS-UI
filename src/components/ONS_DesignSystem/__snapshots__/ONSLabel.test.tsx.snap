// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Label Test matches Snapshot 1`] = `ShallowWrapper {}`;
